encoding/csv/_io.ts

```diff
@@ -0,0 +1,241 @@
+            );
+            quoteError = new ParseError(
+              startLine + 1,
+              lineIndex,
+              col,
+              ERR_QUOTE,
+            );
+            break parseField;
+          }
+        } else if (line.length > 0 || !(await reader.isEOF())) {
+          // Hit end of line (copy all data so far).
+          recordBuffer += line;
+          const r = await reader.readLine();
+          lineIndex++;
+          line = r ?? ""; // This is a workaround for making this module behave similarly to the encoding/csv/reader.go.
+          fullLine = line;
+          if (r === null) {
+            // Abrupt end of file (EOF or error).
+            if (!opt.lazyQuotes) {
+              const col = runeCount(fullLine);
+              quoteError = new ParseError(
+                startLine + 1,
+                lineIndex,
+                col,
+                ERR_QUOTE,
+              );
+              break parseField;
+            }
+            fieldIndexes.push(recordBuffer.length);
+            break parseField;
+          }
+          recordBuffer += "\n"; // preserve line feed (This is because TextProtoReader removes it.)
+        } else {
+          // Abrupt end of file (EOF on error).
+          if (!opt.lazyQuotes) {
+            const col = runeCount(fullLine);
+            quoteError = new ParseError(
+              startLine + 1,
+              lineIndex,
+              col,
+              ERR_QUOTE,
+            );
+            break parseField;
+          }
+          fieldIndexes.push(recordBuffer.length);
+          break parseField;
+        }
+      }
+    }
+  }
+  if (quoteError) {
+    throw quoteError;
+  }
+  const result = [] as string[];
+  let preIdx = 0;
+  for (const i of fieldIndexes) {
+    result.push(recordBuffer.slice(preIdx, i));
+    preIdx = i;
+  }
+  return result;
+}
+
+function runeCount(s: string): number {
+  // Array.from considers the surrogate pair.
+  return Array.from(s).length;
+}
+
+/**
+ * A ParseError is returned for parsing errors.
+ * Line numbers are 1-indexed and columns are 0-indexed.
+ */
+export class ParseError extends Error {
+  /** Line where the record starts*/
+  startLine: number;
+  /** Line where the error occurred */
+  line: number;
+  /** Column (rune index) where the error occurred */
+  column: number | null;
+
+  constructor(
+    start: number,
+    line: number,
+    column: number | null,
+    message: string,
+  ) {
+    super();
+    this.startLine = start;
+    this.column = column;
+    this.line = line;
+
+    if (message === ERR_FIELD_COUNT) {
+      this.message = `record on line ${line}: ${message}`;
+    } else if (start !== line) {
+      this.message =
+        `record on line ${start}; parse error on line ${line}, column ${column}: ${message}`;
+    } else {
+      this.message =
+        `parse error on line ${line}, column ${column}: ${message}`;
+    }
+  }
+}
```

Comment:
wouldn't it make more sense to use syntaxerror? I know it doesnt have `startLine`, but creating new error types is a bit... eh. If deemed necessary to keep, then extend syntaxerror